plugins/net_plugin/include/eosio/net_plugin/net_plugin.hpp

```diff
@@ -16,9 +16,22 @@ namespace eosio {
    using namespace plugin_interface;
 
    struct net_plugin_metrics {
-      runtime_metric num_peers{"net_plugin", "num_peers", 0};
-      runtime_metric num_clients{"net_plugin", "num_clients", 0};
-      runtime_metric dropped_trxs{"net_plugin", "dropped_trxs", 0};
+      runtime_metric num_peers{metric_type::gauge, "num_peers", "num_peers", 0};
+      runtime_metric num_clients{metric_type::gauge, "num_clients", "num_clients", 0};
+      runtime_metric dropped_trxs{metric_type::counter, "dropped_trxs", "dropped_trxs", 0};
+
+      std::vector<std::reference_wrapper<runtime_metric>> metrics;
+
+      void enable(bool enabled)  {_enabled = enabled; }
+      bool enabled() {return _enabled;}
```

Comment:
These methods and `metrics` could be moved to a base class so that any *_metrics struct could derive and get it without duplicating it for all *_metrics structs.